src/coreclr/hosts/unixcoreconsole/CMakeLists.txt

```diff
@@ -31,8 +31,4 @@ if(NOT CLR_CMAKE_PLATFORM_ANDROID)
   )
 endif()
 
-add_dependencies(coreconsole
-    coreclr
```

Comment:
I've tried to merge add_subdirectory(src/coreclr/hosts), but it looked like have some complicate build dependency problem. It's better to exclude the add_subdirectory(src/coreclr/hosts) for NOT DEFINED CLR_CROSS_COMPONENTS_BUILD as you suggested, and remain as another work item that is clean-up build dependencies.